src/iter/mod.rs

```diff
@@ -2341,6 +2341,7 @@ pub trait IndexedParallelIterator: ParallelIterator {
     /// // we should never get here
     /// assert_eq!(1, zipped.len());
     /// ```
+    #[track_caller]
```

Comment:
Since we're changing the panic location, the assertion below needs textual context, something like "iterators must have the same length".